lib/middleware/validate-signature-middleware.js

```diff
@@ -0,0 +1,267 @@
+ * @property {string} method How the key is generated (HMAC)
+ * @property {string} signature The trimmed and standardized signature
+ * @property {string} type Signature type (OT1)
+ */
+
+module.exports = (config, errorResponse, OtDate, promise, signatureOt1) => {
+    var dateWindowFuture, dateWindowPast, signatureHost, signatureLinkDefinition, signatureMethod;
+
+    /**
+     * Splits up the signature information into an object for easier
+     * processing.
+     *
+     * @param {string} signature
+     * @return {Promise.<validateSignatureMiddleware~signatureDescription>}
+     */
+    function parseSignatureInformationAsync(signature) {
+        var cleaned, split;
+
+        cleaned = signature.trim().toUpperCase();
+        split = cleaned.split("-");
+
+        if (split.length !== 4) {
+            return errorResponse.rejectedPromiseAsync("Invalid format of signature identifier.", "9iYNSNlY");
+        }
+
+        return promise.resolve({
+            algorithm: split[2],
+            encoding: split[3],
+            method: split[1],
+            signature: cleaned,
+            type: split[0]
+        });
+    }
+
+
+    /**
+     * Transforms an array of fields similar to ` key="value" ` into
+     * an object (used as a map).
+     *
+     * @param {Array.<string>} chunks
+     * @return {Promise.<Object>}
+     */
+    function parseHeaderKeyValuePairsAsync(chunks) {
+        var key, result, value;
+
+        result = {};
+
+        while (chunks.length) {
+            value = chunks.shift().split("=");
+            key = value.shift().trim();
+            value = value.join("=");
+
+            if (value.charAt(0) === "\"") {
+                // If the value is quoted, we look for the end of the
+                // quotes and may consume more chunks until we get the
+                // end.  The next non-whitespace character after the
+                // quotes must be a semicolon, so if we run out of
+                // chunks and still can't find a quote then we have an
+                // invalid HTTP header value.
+                while (!value.match(/" *$/)) {
+                    if (!chunks.length) {
+                        return errorResponse.rejectedPromiseAsync("Invalid quoting in signature header", "pFDHKguh");
+                    }
+
+                    value += `;${chunks.shift()}`;
+                }
+
+                value = value.substr(1).replace(/" *$/, "");
+            } else {
+                // Trim whitespace at the end.
+                value = value.replace(/ *$/, "");
+            }
+
+            if (result[key]) {
+                return errorResponse.rejectedPromiseAsync(`Duplicated attribute name in Authorized header: ${key}`, "QtlTT61E");
+            }
+
+            result[key] = value;
+        }
+
+        return promise.resolve(result);
+    }
+
+
+    /**
+     * Parses the Authorized header and verifies the signature
+     *
+     * @param {Restify~req} req
```

Comment:
There's an inconsistency between this line and 207. 207 seems a tad more explicit: 
`* @param {Restify~Request} req`